components/script/dom/htmlimageelement.rs

```diff
@@ -1066,31 +1061,36 @@ impl HTMLImageElement {
         let window = window_from_node(self);
         let image_cache = window.image_cache();
         // Step 14
-        let response = image_cache.find_image_or_metadata(
-            img_url.clone().into(),
-            UsePlaceholder::No,
-            CanRequestImages::Yes,
-        );
-        match response {
-            Ok(ImageOrMetadataAvailable::ImageAvailable(_image, _url)) => {
-                // Step 15
+        let response = image_cache.get_image(&img_url, UsePlaceholder::No);
+
+        if let Some(_image) = response {
+            // Step 15
+            return self.finish_reacting_to_environment_change(
+                selected_source,
+                generation,
+                selected_pixel_density,
+            );
+        }
+
+        let cache_result =
+            image_cache.track_image(img_url.clone(), UsePlaceholder::No, CanRequestImages::Yes);
```

Comment:
This is another case where we should be able to merge track_image and get_image calls together.